dev-vcs/bzr/patches/bzr-2.7.0_gcc2.patchset

```diff
@@ -0,0 +1,57 @@
+From 70fab55bb288bfe0976d9afc4c8a58eadb8cd397 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
+Date: Mon, 4 Jun 2018 09:14:57 +0200
+Subject: Use gcc2 zero-length array member
+
+
+diff --git a/bzrlib/diff-delta.c b/bzrlib/diff-delta.c
+index 0801c96..63c19c7 100644
+--- a/bzrlib/diff-delta.c
++++ b/bzrlib/diff-delta.c
+@@ -24,6 +24,12 @@
+ #include <string.h>
+ #include <assert.h>
+ 
++# if defined(__GNUC__) && (__GNUC__ < 3)
++#define FLEX_ARRAY_LEN 0
++#else
+++#define FLEX_ARRAY_LEN
```

Comment:
> Shouldn't it be applied to x86_64 too? Specially the second part of it for the location to the SSL certificates?

Yep, for the second part, i.e. the one related to the `CARootCertificates.pem`.
For the first part, you can also apply it to all arches if you fix a small typo:
```diff
+# if defined(__GNUC__) && (__GNUC__ < 3)
+#define FLEX_ARRAY_LEN 0
+#else
-+#define FLEX_ARRAY_LEN
+#define FLEX_ARRAY_LEN
+#endif
```
(The problem is that you have a `++` instead of a `+` in line 18.)

BTW, you might wish to insert a few spaces to fix the indentation. The result would look like:
```diff
+# if defined(__GNUC__) && (__GNUC__ < 3)
+#  define FLEX_ARRAY_LEN 0
+# else
+#  define FLEX_ARRAY_LEN
+# endif
```

Once you have fixed this, you can rename the patchset to `bzr-2.7.0.patchset` and apply it to all architectures.